compiler/fir/tree/src/org/jetbrains/kotlin/fir/expressions/impl/FirArgumentListWithResolutionInfo.kt

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.expressions.impl
+
+import org.jetbrains.kotlin.fir.declarations.FirValueParameter
+import org.jetbrains.kotlin.fir.expressions.FirExpression
+import org.jetbrains.kotlin.fir.resolve.substitution.ConeSubstitutor
+import org.jetbrains.kotlin.fir.types.ConeKotlinType
+
+interface FirArgumentListWithResolutionInfo {
+    val mapping: LinkedHashMap<FirExpression, FirValueParameter>
+    val paramTypeSubstitutor: ConeSubstitutor?
```

Comment:
Not sure we have a convention, but... everything else is fully named, while this one is abbreviated. Since `Argument` is Argument, not `Arg` or `arg` etc, can we name it `parameterTypeSubstitutor`?

Hmm, actually, it's better to distinguish "type parameter" from value parameter, maybe `typeParameterSubstitutor`?

Reply:
Thanks! Renaming to `typeParameterSubstitutor` sounds good!